Shared/Extensions/URLExtensions.swift

```diff
@@ -366,6 +366,10 @@ extension URL {
     public var safeBrowsingErrorURL: Bool {
         return scheme == "http" && host == "localhost" && path.contains("/errors/SafeBrowsingError.html")
     }
+    
+    public var isSessionRestoreURL: Bool {
+        return scheme == "http" && host == "localhost" && path.contains("/about/sessionrestore")
```

Comment:
Let's abstract the first part of this: `scheme == "http" && host == "localhost"`, as it is used in 3 places now. Maybe a (or something):
```
private var isLocalhost: Bool {
    return scheme == "http" && host == "localhost"
}
```